job-server/src/spark.jobserver/io/DataFileDAO.scala

```diff
@@ -95,13 +95,14 @@ class DataFileDAO(config: Config) {
     out.writeLong(aInfo.uploadTime.getMillis)
   }
 
-  def deleteFile(aName: String) {
+  def deleteFile(aName: String): Boolean = {
     if (aName.startsWith(rootDir) && files.contains(aName)) {
       // only delete the file if it is known to this class,
       // otherwise this could be abused
-      new File(aName).delete
       files -= aName
```

Comment:
I think this should happen after successfully deleting the file. What if the below line throws an exception?